### ISSUE TITLE
Accessibility Unit Testing

### ISSUE THREAD

**Issue by Sabine-Justilien** [Call-for-Code-for-Racial-Justice/Five-Fifths-Voter#209]
### Background on the problem the feature will solve/improved user experience
Five Fifths Voter users with accessibility needs should be able to use our solution without limitations. 

### Describe the solution you'd like
Unit testing is needed to identify if [Level 1 accessibility](https://www.ibm.com/able/requirements/requirements/-) requirements are met. IBM has a robust process for guaranteeing accessibility athttps://www.ibm.com/able/. This is publicly available as a reference for creating, designing, and testing for accessibility needs. 

### Tasks 
- Unit test of 5/5 voter website (can use https://www.ibm.com/able/toolkit/verify) 
- Check against [Level 1 accessibility requirements](https://www.ibm.com/able/requirements/requirements/-) [select Level 1 in Level to Achieve on left hand side filter pane]

### Acceptance Criteria 
- Checklist of Level 1 accessibility requirements met or failed 
- Path forward to meeting 75% of the 21 requirements under Level 1

**Comment by Sabine-Justilien** [comment]
[Accessibility_Report-five-fifths voter.xlsx](https://github.com/Call-for-Code-for-Racial-Justice/Five-Fifths-Voter/files/7173478/Accessibility_Report-five-fifths.voter.xlsx)

**Comment by davidnixon** [comment]
Fiewor - these are the 2 most critical issues (copied from the spreadsheet attached above):

- [ ]     The element with "menuitem" role is not contained in or owned by an element with one of the following ARIA roles: "group, menu, menubar"
- [ ]      The 'id' "user-panel" specified for the ARIA property 'aria-controls' value is not valid

**Comment by davidnixon** [comment]
Hmm I actually see different violations compared to what is attached in the spreadsheet. I see:

- [ ] Content is not within a landmark element
    `<h1 class="hero__heading">`
    **What to do**
    Enclose the content within an HTML5 sectioning element with an implicit landmark role (e.g.` <main>Content</main>`);
Or, use the "role" attribute on an enclosing element to specify an appropriate ARIA landmark value (e.g. `<div role="contentinfo">...</div>`).

- [ ] Content is not within a landmark element
    `<a class="bx--feature-card__link" href="#/whyvote">`
    **What to do**
    Enclose the content within an HTML5 sectioning element with an implicit landmark role (e.g.` <main>Content</main>`);
Or, use the "role" attribute on an enclosing element to specify an appropriate ARIA landmark value (e.g. `<div role="contentinfo">...</div>`).

- [ ] The 'id' "user-panel" specified for the ARIA property 'aria-controls' value is not valid
    **What to do**
    - Update the ARIA property value with the unique id value of a valid visible element.

    For example:
    ```
    <form action="" method="post">
      <label for="ln1">Last Name</label>
      <input type="text" name="lastname" id="ln1" value="enter last name" title="enter last name" aria-describedby="ln2">
    </form>
    <div id="ln2" role="tooltip">Last Name must be alpha only characters</div>
    ```

- [ ] Multiple elements with "navigation" role do not have unique labels
    **What to do**
    - For each element with a "nav" role, either set the value of the aria-labelledby attribute to the id value of any visible text that appropriately and uniquely describes that navigation region;
    - Or, use the aria-label attribute to add an invisible, unique, non-empty label to an element with the "nav" role.

    For example:
    ```
    <h2 id="weather">Weather portlet</h2>
    <nav aria-labelledby="weather">
        ...
    </nav>
    <nav aria-label="Stock ticker portlet">
            ...
    </nav>
   ```

**Comment by davidnixon** [comment]
I cannot reproduce the **The element with "menuitem" role is not contained in or owned by an element with one of the following ARIA roles: "group, menu, menubar"** issue that was originally reported. Maybe it has been fixed since the report was run. I see that the all "menuitem" are in a "menubar".

**Comment by Fiewor** [comment]
Alright. Let me take a close look

**Comment by davidnixon** [comment]
@upkarlidder this is the issue we discussed on the scrum but the title I think may be off? I think we discussed just getting fixes for the existing accessibility issues but this one calls for unit tests.

The unit tests is a bigger ask so I think we continue tracking accessibility fixes here and, once complete, work on a unit test. Maybe that is your intent too. Just making sure we are aligned.

**Comment by upkarlidder** [comment]
I agree @davidnixon. Let's change the title to `Identify and fix  Level 1 accessibility issues`. And I can create a new one for adding accessibility testing as a requirement in CI/CD pipeline. Would that work?

FYI @Sabine-Justilien @demilolu

**Comment by Fiewor** [comment]
> * [ ]  Multiple elements with "navigation" role do not have unique labels
>   **What to do**
>   
>   * For each element with a "nav" role, either set the value of the aria-labelledby attribute to the id value of any visible text that appropriately and uniquely describes that navigation region;
>   * Or, use the aria-label attribute to add an invisible, unique, non-empty label to an element with the "nav" role.
>   
>   For example:
>   ```
>   <h2 id="weather">Weather portlet</h2>
>   <nav aria-labelledby="weather">
>       ...
>   </nav>
>   <nav aria-label="Stock ticker portlet">
>           ...
>   </nav>
>   ```
>   
>   
>       
>         
>       
>   
>         
>       
>   
>       
>     
>   **Violations are in this code (AFAIK)**
>   
>   * ui/src/components/AppHeader/AppHeader.vue

I'm having a bit of an issue with this one. I can't quite see any element with a role in the code you mentioned. I only see one with an id of "side-nav". Should that be the only one I just give an aria-label?

**Comment by demilolu** [comment]
@upkarlidder @davidnixon can you help here?

**Comment by davidnixon** [comment]
@Fiewor I see why you are stuck. There seems to be some leftovers here from a different navigation that is no longer available.

This code has a label already but i looks like just a placeholder. I think that one should be something like "main menu"
https://github.com/Call-for-Code-for-Racial-Justice/Five-Fifths-Voter/blob/efbe06b3c8147c6d0d5b2182e5482582da688861/ui/src/components/AppHeader/AppHeader.vue#L10


And this code has a nav container but it is never visible because `areLeftPanels` is always false
https://github.com/Call-for-Code-for-Racial-Justice/Five-Fifths-Voter/blob/efbe06b3c8147c6d0d5b2182e5482582da688861/ui/src/components/AppHeader/AppHeader.vue#L29-L48

I suggest this:

- [ ] Update the aria label on `cv-header-nav`
- [ ] Delete the entire template `v-slot:left-panels` and its content
    ```vue
    <template v-slot:left-panels v-if="areLeftPanels">
    ...
    </template>
    ```
- [ ] Since `areLeftPanels` is never used, remove it from the components data
    ```vue
  data() {
    return {};
  }
    ```

**Comment by Fiewor** [comment]
Alright. Thank you. I was stuck. I'll do that.

**Comment by demilolu** [comment]
are we set to close this :)? I'm also looking for your pull request @Fiewor where is it :)

**Comment by Fiewor** [comment]
I have opened a pull request @demilolu  but I'll have to close it and send another one because @davidnixon has pointed out some errors that I have to correct

**Comment by krook** [comment]
@Fiewor FYI you should be able to push new commits to an open pull request. No need to close/reopen.

**Comment by upkarlidder** [comment]
+1 @krook 

@Fiewor if you commit to the same branch in the remote, it pushes to the same PR as a new commit. Your PR can have multiple commits. GitHub is smart to recognize that you are working on the same branch. If you deleted the branch locally, recreate with the same name and push.

**Comment by Fiewor** [comment]
Oh. I didn't know that. Thank you. I'll do this .
